File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.py
class Solution:
  def maximumNumberOfStringPairs(self, words: List[str]) -> int:
    hash_set = set()
    
    ans = 0
    for word in words:
      hash_key = tuple(sorted(word))
      if hash_key in hash_set:
        hash_set.remove(hash_key)
        ans += 1
      else:
        hash_set.add(hash_key)
    
    return ans 
        
      
        

File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.cpp
class Solution {
public:
  int maximumNumberOfStringPairs(vector<string>& words) {
    unordered_set<string> hash_set;
    
    int ans = 0;
    for (auto& word: words) {
      sort(begin(word), end(word));
      if (hash_set.find(word) != end(hash_set) ) {
        hash_set.erase(word);
        ans++;
      } else {
        hash_set.insert(word);
      }
    }
    
    return ans;
  }
};